java/adaptersdk-schema-java-lib/src/test/java/io/github/fileanalysissuite/adaptersdk/schema/MapSchemaObjectBuilder.java

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright 2022-2023 Micro Focus or one of its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.fileanalysissuite.adaptersdk.schema;
+
+import io.github.fileanalysissuite.adaptersdk.schema.builders.JsonBuilder;
+import io.github.fileanalysissuite.adaptersdk.schema.builders.SchemaObjectBuilder;
+import io.github.fileanalysissuite.adaptersdk.schema.model.Field;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public final class MapSchemaObjectBuilder implements SchemaObjectBuilder
+{
+    private final Map<String, Object> document;
+
+    public MapSchemaObjectBuilder(final Map<String, Object> document)
+    {
+        this.document = document;
+    }
+
+    @Override
+    public void clearField(final Field field)
+    {
+        document.remove(field.getFieldName());
+    }
+
+    @Override
+    public void setStringFieldValue(final Field field, final String value)
+    {
+        document.put(field.getFieldName(),
+                     field.isMultivalue() ? Collections.singletonList(value) : value);
+    }
+
+    @Override
+    public void setStringFieldValue(final Field field, final String... values)
+    {
+        document.put(field.getFieldName(), Arrays.asList(values));
+    }
+
+    @Override
+    public void setStringFieldValue(final Field field, final List<String> values)
+    {
+        document.put(field.getFieldName(), values);
+    }
+
+    @Override
+    public void addStringFieldValue(final Field field, final String value)
+    {
+        Object currentValue = document.get(field.getFieldName());
+        if (currentValue == null) {
+            currentValue = new ArrayList<String>();
+        }
+        document.put(field.getFieldName(), ((ArrayList<String>) currentValue).add(value));
```

Comment:
This line is generating a warning in the log and the warning is absolutely correct.  This code would not be safe in many circumstances.  For example if the user was to call the `setStringFieldValue(final Field field, final String value)` overload first before calling this `addStringFieldValue(final Field field, final String value)` overload, then this code would not work.